orangecontrib/spectroscopy/widgets/owoverlay.py

```diff
@@ -0,0 +1,281 @@
+        
+        gui.rubber(self.controlArea)
+        gui.auto_commit(self.controlArea, self, "autocommit", "Send Data")
+
+        self.imageplot.color_cb.activated.connect(self.commit.deferred)
+        self.imageplot._threshold_low_slider.sliderReleased.connect(self.commit.deferred)
+        self.imageplot._threshold_high_slider.sliderReleased.connect(self.commit.deferred)
+        self.imageplot._level_high_le.editingFinished.connect(self.commit.deferred)
+        self.imageplot._level_low_le.editingFinished.connect(self.commit.deferred)
+        self._init_attr_values(self.data)
+
+    def _attr_changed(self):
+        self._update_attrs()
+        self.commit.deferred()
+
+    def _init_attr_values(self, data):
+        domain = data.domain if data is not None else None
+        self.feature_value_model.set_domain(domain)
+        self.attr_value = (
+            self.feature_value_model[0] if self.feature_value_model else None
+        )
+        self.xy_model.set_domain(domain)
+        self.attr_x = self.xy_model[0] if self.xy_model else None
+        self.attr_y = self.xy_model[1] if len(self.xy_model) >= 2 else self.attr_x
+
+    def _init_interface_data(self, data):
+        self._init_attr_values(data)
+        self.imageplot.init_interface_data(data)
+
+    def _update_feature_value(self):
+        self.redraw_data()
+
+    def _update_attrs(self):
+        self.imageplot.attr_x = self.attr_x
+        self.imageplot.attr_y = self.attr_y
+        self.redraw_data()
+    
+    def redraw_data(self):
+        self.imageplot.update_view()
+        self.commit.deferred()
+    
+    def set_vimage(self, data):
+        # Extract the image scales
+        if data is not None:
+            try:
+                hypercube, ls = get_ndim_hyperspec(
+                        data, (self.imageplot.attr_x,self.imageplot.attr_y)
+                    )
+            except Exception as e:
+                self.Error.invalid_axis(str(e))
+                return
+            
+            width = np.abs(ls[0][1]-ls[0][0])
+            height = np.abs(ls[1][1]-ls[1][0])
+            xres = width/hypercube.shape[:2][0]
+            yres = height/hypercube.shape[:2][1]
+            posx = ls[0][0]
+            posy = ls[1][0]
+
+            # Extract the image from imageplot so the coloring is done
+            im = self.imageplot.img.qimage
+            if im is not None:
+                buffer = QBuffer()
+                buffer.open(QBuffer.ReadWrite)
+                im.save(buffer, "PNG")
+                pil_im = Image.open(io.BytesIO(buffer.data()))
+                pil_im = pil_im.transpose(Image.FLIP_TOP_BOTTOM)
+                img_bytes = io.BytesIO()
+                pil_im.save(img_bytes, format='PNG')
+
+                # Need to modify the position and the scale since visual imageplot 
+                # place the corner of the pixel to the given position
+                vimage = ConstantBytesVisibleImage(name="External Image",
```

Comment:
Because images should be added, an unused name should be chosen here (Orange has functions for choosing new names)